pkg/node/chain.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p/libp2p"
+	"github.com/ethersphere/bee/pkg/settlement/swap"
+	"github.com/ethersphere/bee/pkg/settlement/swap/chequebook"
+	"github.com/ethersphere/bee/pkg/settlement/swap/swapprotocol"
+	"github.com/ethersphere/bee/pkg/settlement/swap/transaction"
+	"github.com/ethersphere/bee/pkg/storage"
+)
+
+const (
+	maxDelay = 1 * time.Minute
+)
+
+// InitChain will initialize the Ethereum backend at the given endpoint and
+// set up the Transacton Service to interact with it using the provided signer.
+func InitChain(
+	ctx context.Context,
+	log logging.Logger,
+	stateStore storage.StateStorer,
+	endpoint string,
+	signer crypto.Signer,
+) (
+	backend *ethclient.Client,
```

Comment:
In this case, some of the named returns help to understand the semantics behind types like chainID. But, it is ok with only a comment and no named returns.

Reply:
I agree that using named returns helps with code readability under some circumstances, but I find the setting of the variable and a blank return just not very nice, and makes code branching to become non-linear, plus leaves a lot of room for error (forgot to return?)

Reply:
Is there general style consensus in the bee codebase? I believe I've found examples of both cases. I can make the change (quite easily) but it would be prudent to have a style guide for future work.

Reply:
We use named returns very rarely:
- In the case that there are quite a lot of return values (more than a few, in cases where you cannot infer which kind of object you're getting. e.g. `func someFunc() (int, int, int, int)` would probably be more readable as `func someFunc() (height int, width int, topleft int, bottomright int)`
- In cases where the compiler optimization that it provides actually gives benefit